include/swift/AST/Decl.h

```diff
@@ -2321,10 +2321,8 @@ class ValueDecl : public Decl {
   /// dynamic methods on generic classes (see above).
   bool isNativeMethodReplacement() const;
 
-  bool isEffectiveLinkageMoreVisibleThan(ValueDecl *other) const {
-    return (std::min(getEffectiveAccess(), AccessLevel::Public) >
-            std::min(other->getEffectiveAccess(), AccessLevel::Public));
-  }
+  /// Returns if this declaration has more visible formal access than 'other'.
```

Comment:
Should the comment and/or name still indicate that this takes effective visibility into account?